api/firmware/btc.go

```diff
@@ -430,24 +430,75 @@ func (device *Device) BTCSignMessage(
 		return nil, 0, nil, UnsupportedError("9.2.0")
 	}
 
+	supportsAntiklepto := device.version.AtLeast(semver.NewSemVer(9, 5, 0))
+	var hostNonceCommitment *messages.AntiKleptoHostNonceCommitment
+	var hostNonce []byte
```

Comment:
This looks more like C. No need to declare it upfront. Easier to read:

```go
if supportsAntiklepto {
  n, err := randomBytes(32)
  if err != nil {
    return ...
  }
  hostNonceCommitment = &messages.AntiKleptoHostNonceCommitment{
    Commitment: antikleptopHostCommit(n),
  }
}
```
or you could do it after the base of the request, eliminating `var hostNonceCommitment ...`:

```go
pbreq := &messages.BTCRequest{
  Request: &messages.BTCRequest_SignMessage{...},
}
if supportsAntiklepto {
  ...
  pbreq.Request.SignMessage.HostNonceCommitment: &messages.AntiKleptoHostNonceCommitment{
    Commitment: antikleptopHostCommit(n),
  }
}
```

Reply:
And the same in ETH, I guess.

Reply:
None of the two approaches work unfortunately. First one: hostNonce is used again below for verification.

The second approach does not work (easily) because half the fields are interface{} types. I tried and found it easier to go the other way by declaring it upfront than to make a bunch of type assertions below.

Reply:
Ah yes of course, the device needs the raw nonce in the end.